api/events/types.go

```diff
@@ -17,12 +17,33 @@ import (
 )
 
 type LogMeta struct {
-	BlockID        thor.Bytes32 `json:"blockID"`
-	BlockNumber    uint32       `json:"blockNumber"`
-	BlockTimestamp uint64       `json:"blockTimestamp"`
-	TxID           thor.Bytes32 `json:"txID"`
-	TxOrigin       thor.Address `json:"txOrigin"`
-	ClauseIndex    uint32       `json:"clauseIndex"`
+	BlockID         thor.Bytes32     `json:"blockID"`
+	BlockNumber     uint32           `json:"blockNumber"`
+	BlockTimestamp  uint64           `json:"blockTimestamp"`
+	TxID            thor.Bytes32     `json:"txID"`
+	TxOrigin        thor.Address     `json:"txOrigin"`
+	ClauseIndex     uint32           `json:"clauseIndex"`
+	ExtendedLogMeta *ExtendedLogMeta `json:"extendedLogMeta,omitempty"`
```

Comment:
Can this just be 

```
ClauseIndex uint32
TxIndex *uint32
LogIndex *uint32
```

Reply:
And this should be also applied to `TransferLog`

Reply:
Yes, the structure can be flattened, I was thinking this way is a bit tidier to gather the optional params separately (it can also be useful in case we need to add more fields). But let me know if you think in this case is better to have them with together with the other fields

Reply:
IMO flattened is better too